src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -156,15 +163,58 @@ public void setPerson(Person target, Person editedPerson) {
         addressBook.setPerson(target, editedPerson);
     }
 
+    @Override
+    public void sortPerson(Index index) {
+        Comparator<Person> nameComparator = new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return o1.getName().fullName.compareToIgnoreCase(o2.getName().fullName);
+            }
+        };
+
+        Comparator<Person> addressComparator = new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return o1.getAddress().value.compareToIgnoreCase(o2.getAddress().value);
+            }
+        };
+
+        Comparator<Person> tagComparator = new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                System.out.println(o2.getTags().size());
+                if (o1.getTags().size() == 0 && o2.getTags().size() != 0) {
+                    return 1;
+                }
+                if (o1.getTags().size() != 0 && o2.getTags().size() == 0) {
+                    return -1;
+                }
+                if (o1.getTags().size() == 0 && o2.getTags().size() == 0) {
+                    return o1.getName().fullName.compareToIgnoreCase(o2.getName().fullName);
+                }
+                return o1.getTags().iterator().next().tagName
+                        .compareToIgnoreCase(o2.getTags().iterator().next().tagName);
+            }
+        };
```

Comment:
The three comparators should probably declared as static constants within the `SortCommand` class itself?